src/shims/windows/fs.rs

```diff
@@ -204,20 +202,46 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
             );
         }
 
+        // This is racy, but there doesn't appear to be an std API that both succeeds if a file
+        // exists but tells us it isn't new. Either we accept racing one way or another, or we
+        // use an iffy heuristic like file creation time. This implementation prefers to fail in the
+        // direction of erroring more often.
+        let exists = file_name.exists();
```

Comment:
This comment is too early, since one doesn't know yet that this is used for. IMO it'd be better placed at the `if exists` below.

(Also, no need to `let`-bind, I don't feel like this makes things much clearer.)